### ISSUE TITLE
Add a Trigger Simulation for mid.gz files

### ISSUE THREAD

**Issue by slwatkins** [ucbpylegroup/RQpy#61]
We will need to simulate the trigger of the DCRC to understand its effect on a DM limit, so a simulation function should be written that takes in data/template/PSD, correctly sets up the FIR filter, and outputs the trigger amplitude.

**Comment by slwatkins** [comment]
This is nearing completion on the `feature/trigsim` branch. I've added the wrapper code for Jon's trigger simulation code (which I refactored into a package called `trigsim`). I've also added it to the RQ processing. There are still some things I'm thinking about:

- I'm not currently actually using the trigger, as I just take the maximum amplitude within the FIR filtered trace and output it. Thus, I can actually output values less than the trigger. Does this matter? Maybe not, as we know what the trigger was set at and can simply make a cut on the data points that have simulated FIR amplitudes less than the specified threshold.
- When running on real data, should I be checking the trigger type to make sure it's not being run on randoms? Maybe it doesn't matter, since we can simply ignore the trigger amplitude for randoms. This would matter for computational reasons though.
- Should I care about the time of the trigger? I currently am simply taking the largest amplitude anywhere in the trace. This might matter for simulated data, as if there is a large real pulse in the trace with larger amplitude than the simulated pulse, then the max amplitude will not correspond to the amplitude of the simulated pulse. Maybe this is a pulse sim problem though.

Anyways, I'll probably have more commits and merge this week.
